modules/dbparser/dbparser-grammar.ym

```diff
@@ -126,6 +127,7 @@ grouping_by_opts
 
 grouping_by_opt
 	: KW_KEY '(' template_content ')'                       { grouping_by_set_key_template(last_parser, $3); log_template_unref($3); }
+        | KW_SORT_KEY '(' string ')'				{ grouping_by_set_sort_key(last_parser, log_msg_get_value_handle($3)); free($3); }
```

Comment:
Just one question:
`key()` receives a template, for example: `key("${MESSAGE}")`, but `sort-key()` wants a string, the name of the key: `sort-key("PROGRAM")`.

Wouldn't it be better if it was unified?
Using NVHandles is much cheaper than formatting templates, that's an advantage.

However, `key()` as a template supports multiple keys: `key("$field1 $field2")`. The same might be possible for sorting keys.

Reply:
I saw the inconsistency, however this is much faster. The compare function needs to be called a lot more times than the key function, and I was thinking this should be enough.

Maybe a sort-key() and a sort-value() could work?

Or we might be able to recognize if the parameter is a template or just one value.

Reply:
I saw the inconsistency, however this is much faster. The compare function needs to be called a lot more times than the key function, and I was thinking this should be enough.

Maybe a sort-key() and a sort-value() could work?

Or we might be able to recognize if the parameter is a template or just one value.

Reply:
Both ideas sound good to me. I guess both can be implemented later without breaking anything, so I don't want to block this PR.

Reply:
I have now implemented template support for sort-key(), in a way that "trivial" templates are
using a faster code-path.